W-14038153 usage reports (#206)
mulesoft/docs-general: modules/ROOT/pages/_partials/mule-runtime-metrics.adoc

// tag::intro[]
Mule runtime reports track your use of Mule messages, Mule flows, and data throughput. 

To track Mule message usage, the runtime report counts the number of times a Mule event source triggers a Mule message. You can view the number of these messages in a given day or month by business group, environment, and application. 

> To calculate usage, MuleSoft meters and aggregates the number of messages on a daily and monthly basis. After a message is triggered, the report does not track changes to the message because the message is processed within the application’s flows.

Reviewer: ```suggestion
To calculate usage, MuleSoft meters and aggregates the number of messages on a daily and monthly basis. After a message is triggered, the report doesn’t track changes to the message because the message is processed within the application’s flows.
```

Reviewer: For "daily and monthly basis", can you just say "daily and monthly"?

> A Mule flow is a sequence of logical operations configured within the XML `<flow/>` element of a Mule application. The runtime report tracks the Mule flows within a deployed and running Mule application that contains a Mule event source or APIkit route requests. This report doesn't track other Mule flows. You can view the maximum number of Mule flows in a given day or month in the usage report. 

Reviewer: Do you need "deployed and"? Does an app need to be deployed to run?

To calculate usage, MuleSoft tracks the number of Mule flows for all business groups, environments, and applications. The maximum number of Mule flows within a day or month is identified based on the peak hour across the day or month. For the detailed breakdown, MuleSoft shows the peak hour usage per business group, environment, and application. 

> Data throughput is all the network I/O bytes produced by the infrastructure that starts and runs the Mule runtime engine that runs a Mule application. You can view the total amount of GB usage in a given day or month by business group, environment, and application. 

Reviewer: Any way to simplify? 
Data throughput is the network I/O bytes produced by the infrastructure that runs the Mule runtime engine running the Mule application.

> To calculate usage, MuleSoft tracks GB usage and aggregates the total GBs on a daily and monthly basis.

Reviewer: ```suggestion
To calculate usage, MuleSoft tracks GB usage and aggregates the total daily and monthly GBs.
```

Reviewer: (Does that change the meaning?)

Reviewer 2: i don't think so.

// end::intro[]

== Requirements for Mule Runtime to Report Usage

// tag::supportedRuntimes[]

include::partial$usage-runtime.adoc[]

> You must redeploy your apps with a supported version, either manually or during the monthly patching cycle. The monthly patching cycle is applicable only for CloudHub and CloudHub 2.0 applications. 

Reviewer: Also applies to RTF, but not sure if that is relevant here.

> To xref:cloudhub-2::ch2-update-apps.adoc#roll-back-to-a-previous-successful-configuration[roll back] to a previous successful configuration of existing applications that use a Mule version that usage reports don't support, you must choose a compliant version before deploying changes. After deployment, a new configuration with the previously chosen settings and a supported runtime is created.

Reviewer: The first sentence is a doozy - maybe split into two? Should it be singular "existing app"?

// end::supportedRuntimes[]

// tag::cards[]

Total Data Throughput::
The sum of data in gigabytes (GB) transmitted across all Mule apps and business groups during the selected time period. For more information about how this metric is billed, see xref:usage-metrics.adoc#data-throughput[data throughput].  

Total Mule Messages:: 
The sum of all Mule messages transmitted across all Mule apps and business groups during the selected time period. For more information about how this metric is billed, see xref:usage-metrics.adoc#mule-messages[Mule messages].  

Max Flows::
The maximum number of concurrent flows across all Mule apps and business groups during the selected time period. For more information about how this metric is billed, see xref:usage-metrics.adoc#mule-flows[Mule flows].  

// end::cards[]

// tag::table[]
Business Group:: 
The business group in which a Mule app is deployed.

Environment:: 
The environment in which a Mule app is deployed.

Application::
The name of the Mule application.

Deployment:: 

> The runtime plane to which the Mule app is deployed. CloudHub (abbreviated as CH), CloudHub 2.0 (abbreviated as CH2), and Runtime Fabric (abbreviated as RTF) are supported. 

Reviewer: Are those abbreviations in the UI?

Reviewer 2: @hannanelson I had ASSumed so, but I just went and checked the UI and the only abbreviation I see there is RTF. I think I'd better leave it though. There might be some dynamic part of the usage reports I just don't see because I don't actually have any.

Flows:: 
The sum of flows in the Mule app. This number is calculated by multiplying flows by the number of workers (CloudHub) or replicas (CloudHub 2.0). For more information about how this metric is billed, see xref:usage-metrics.adoc#mule-flows[Mule flows].

Messages::
The sum of inbound and outbound Mule messages in the Mule app. For more information about how this metric is billed, see xref:usage-metrics.adoc#mule-messages[Mule messages].  

Data Throughput::
The sum of inbound and outbound data in gigabytes (GB) transmitted by the Mule app. For more information about how this metric is billed, see xref:usage-metrics.adoc#data-throughput[data throughput].  

Number of Workers::

> The sum of all workers (CloudHub) or replicas (CloudHub 2.0) of a Mule app. Apps that do not have workers or replicas display a value of `N/A` in this column.

Reviewer: ```suggestion
The sum of all workers (CloudHub) or replicas (CloudHub 2.0) of a Mule app. Apps that don’t have workers or replicas show a value of `N/A` in this column.
```

// image

// end::billableFlows[]

// tag::nonBillableFlows[]
[[non-billable-flows]]

> === Mule Flows that Don't Count Against Your Anypoint Platform Package Allotment 

Reviewer: ```suggestion
=== Mule Flows That Don't Count Against Your Anypoint Platform Package Allotment 
```

> Mule Flows that don't have an event source and aren't used for the routing of APIkit requests are not charged against your Anypoint Platform package allotment. These are Mule flows primarily used to modularize code.

Reviewer: ```suggestion
Mule flows that don't have an event source and aren't used for routing APIkit requests aren’t charged against your Anypoint Platform package allotment. These Mule flows are primarily used to modularize code.
```
